project/src/main/webapp/script.js

```diff
@@ -12,19 +12,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.import java.io.IOException;
 
+
 /**
  * Fetches recommended places from the 'query' servlet, and switches from the query form to the
  * results elements in order to display them to the user.
  */
 function fetchFromQuery() {
   document.getElementById('query-form').style.display = 'none';
   document.getElementById('results').style.display = 'block';
+  document.getElementById('map-container').style.display = 'none';
+  document.getElementById('feedback-box').style.display = 'none';
+  const map = createMap();
   const placesDiv = document.getElementById('place');
-  fetch('/query').then(response => response.json()).then((places) => {
+  fetch('/query')
+  .then(response => response.json())
+  .then((places) => {
     places.forEach((singlePlace) => {
       placesDiv.appendChild(createPlaceElement(singlePlace));
-    });
-  });
+      addPlaceMarker(map, singlePlace)
+      })
+    })
+  .then(() => {document.getElementById('waiting-message').style.display = 'none'})
+  .then(() => {document.getElementById('map-container').style.display = 'block'})
+  .then(() => {document.getElementById('feedback-box').style.display = 'block'});
```

Comment:
Why are these all in separate lambdas? 
I think that all of these should go under one method, something like - "displayMapAfterResults". 
I think that it can even include the entire logic after the 2nd "then".

Reply:
Yes it really should be one method, I didn't think about it. I changed it.